File: app/src/main/java/com/android/learn/mcontract/TodoEditContract.java
package com.android.learn.mcontract;

import com.android.base.mmodel.BaseData;

import java.util.Map;

/**
 * Created by gaolei on 2018/6/18.
 */

public class TodoEditContract {

    public interface Presenter {

        void updateTodo(int id, Map<String, Object> map);

    }

    public interface View {


        void showUpdateTodo(BaseData todoData);
    }
}

File: app/src/main/java/com/android/learn/mcontract/TodoEditContract.kt
package com.android.learn.mcontract

import com.android.base.mmodel.BaseData
import com.android.base.mview.BaseView

/**
 * Created by gaolei on 2018/6/18.
 */

class TodoEditContract {

    interface Presenter {

        fun updateTodo(id: Int, map: Map<String, Any>)

    }

    interface View : BaseView {


        fun showUpdateTodo(todoData: BaseData)
    }
}
